src/cmd/sdf_main.cc

```diff
@@ -0,0 +1,203 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <string>
+
+#include <gz/utils/cli/CLI.hpp>
+#include <gz/utils/cli/GzFormatter.hpp>
+
+#include "sdf/config.hh"
+#include "gz.hh"
+
+//////////////////////////////////////////////////
+/// \brief Enumeration of available commands
+enum class SdfCommand
+{
+  kNone,
+  kSdfCheck,
+  kSdfDescribe,
+  kSdfGraph,
+  kSdfPrintInertialStats,
+  kSdfPrint
+};
+
+//////////////////////////////////////////////////
+/// \brief Structure to hold all available topic options
+struct SdfOptions
+{
+  /// \brief Command to execute
+  SdfCommand command{SdfCommand::kNone};
+
+  /// \brief Path to the SDFormat file to print or check
+  std::string filepath{""};
+
+  /// \brief Version of the SDFormat specification to describe
+  std::string version{""};
+
+  /// \brief Type of SDFormat graph to print:
+  /// * "frame" for FrameAttachedToGraph
+  /// * "pose" for PoseRelativeToGraph
+  std::string graphType{""};
+
+  /// \brief When nonzero, preserve included tags when printing converted arg
+  /// (does not preserve merge-includes)
+  int preserveIncludes{0};
+
+  /// \brief When nonzero, print pose rotations in degrees.
+  int degrees{0};
+
+  /// \brief When nonzero, auto-computed inertial values will be printed.
+  int expandAutoInertials{0};
+
+  /// \brief Output stream precision for floating point numbers.
+  std::optional<int> precision;
+
+  /// \brief If set, printed rotations are snapped to specified degree
+  /// intervals.
+  std::optional<int> snapToDegrees;
+
+  /// \brief Printed rotations are snapped if they are less than this specified
+  /// tolerance.
+  double snapTolerance{0.01};
+};
+
+//////////////////////////////////////////////////
+/// \brief Callback fired when options are successfully parsed
+void runSdfCommand(const SdfOptions &_opt)
+{
+  switch(_opt.command)
+  {
+    case SdfCommand::kSdfCheck:
+      cmdCheck(_opt.filepath.c_str());
+      break;
+    case SdfCommand::kSdfDescribe:
+      cmdDescribe(_opt.version.c_str());
+      break;
+    case SdfCommand::kSdfGraph:
+      cmdGraph(_opt.graphType.c_str(), _opt.filepath.c_str());
+      break;
+    case SdfCommand::kSdfPrintInertialStats:
+      cmdInertialStats(_opt.filepath.c_str());
+      break;
+    case SdfCommand::kSdfPrint:
+      cmdPrint(_opt.filepath.c_str(), _opt.degrees, *_opt.snapToDegrees,
+               _opt.snapTolerance, _opt.preserveIncludes, *_opt.precision,
+               _opt.expandAutoInertials);
```

Comment:
What would happen if the optional parameters are not specified by the user? Would calling `operator*` throw? This might be where the windows warning is coming from. It might be best to just initialize these with default values instead of using `std::optional`.